src/main/java/com/epam/reportportal/auth/config/SamlServiceProviderConfiguration.java

```diff
@@ -0,0 +1,150 @@
+package com.epam.reportportal.auth.config;
+
+import com.epam.reportportal.auth.util.CertificationUtil;
+import com.epam.ta.reportportal.database.dao.ServerSettingsRepository;
+import com.epam.ta.reportportal.database.entity.settings.ServerSettings;
+import org.opensaml.saml.saml2.core.NameID;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.saml.key.SimpleKey;
+import org.springframework.security.saml.provider.SamlServerConfiguration;
+import org.springframework.security.saml.provider.config.NetworkConfiguration;
+import org.springframework.security.saml.provider.config.RotatingKeys;
+import org.springframework.security.saml.provider.service.config.ExternalIdentityProviderConfiguration;
+import org.springframework.security.saml.provider.service.config.LocalServiceProviderConfiguration;
+import org.springframework.security.saml.saml2.metadata.NameId;
+
+import java.security.PrivateKey;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static java.util.Base64.getEncoder;
+
+/**
+ * SAML service provider configuration based on application settings
+ *
+ * @author Yevgeniy Svalukhin
+ */
+@Configuration
+public class SamlServiceProviderConfiguration {
+
+    @Value("${rp.auth.saml.base-path}")
+    private String basePath;
+
+    @Value("${rp.auth.saml.key-alias}")
+    private String keyAlias;
+
+    @Value("${rp.auth.saml.key-password}")
+    private String keyPassword;
+
+    @Value("${rp.auth.saml.key-store}")
+    private String keyStore;
+
+    @Value("${rp.auth.saml.key-store-password}")
+    private String keyStorePassword;
+
+    private ServerSettingsRepository serverSettingsRepository;
+
+    public SamlServiceProviderConfiguration(ServerSettingsRepository serverSettingsRepository) {
+        this.serverSettingsRepository = serverSettingsRepository;
+    }
+
+    @Bean(name = "spConfiguration")
+    public SamlServerConfiguration samlServerConfiguration() {
+        return new SamlServerConfiguration().setServiceProvider(serviceProviderConfiguration())
+                .setNetwork(networkConfiguration());
+    }
+
+    private NetworkConfiguration networkConfiguration() {
+        return new NetworkConfiguration().setConnectTimeout(5000)
+                .setReadTimeout(10000);
+
+    }
+
+    private LocalServiceProviderConfiguration serviceProviderConfiguration() {
+        LocalServiceProviderConfiguration serviceProviderConfiguration = new LocalServiceProviderConfiguration();
+        serviceProviderConfiguration.setSignRequests(true)
+                .setWantAssertionsSigned(true)
+                .setEntityId("report.portal.sp.id")
```

Comment:
The same. let it be application.yml's parameters

Reply:
Has been extracted into yml's config